index.js

```diff
@@ -13,45 +13,72 @@ export default class AddonInstaller extends Plugin {
-                      await vizality.manager[pluginsOrThemes].uninstall(
-                        addonID
-                      );
-                    } else {
-                      await vizality.manager[pluginsOrThemes].install(addonURL);
-                    }
-                  }}
-                />
-              </ContextMenu.Group>
-            </>
-          );
-        }
+        if (channelID === '755005584322854972') {
+          if (vizality.manager.plugins.isEnabled('00pccompat')) {
+            const addonURL = args[0].message?.content.match(/https?:\/\/github\.com\/[a-zA-Z0-9-]+\/[a-zA-Z0-9-]+\/?/m)[0];
+            const addonID = addonURL.split('/').pop().toLowerCase();
+            const addonIsInstalled = vizality.manager.plugins.keys.includes(addonID);
+            res.props.children.push(
+              <>
+                <ContextMenu.Separator />
+                <ContextMenu.Group>
+                  <ContextMenu.Item
+                    label={`${addonIsInstalled ? 'Uninstall' : 'Install'} ${'Plugin'
+                    }`}
+                    id="addon-installer"
+                    action={async () => {
+                      if (addonIsInstalled) {
+                        await vizality.manager.plugins.uninstall(
+                          addonID
+                        );
+                      } else {
+                        await vizality.manager.plugins.install(addonURL);
+                      }
+                    }}
+                  />
+                </ContextMenu.Group>
+              </>
+            );
+          }
+        } else {
+          const isPlugin = channelID === '753291447523868753' && true;
+
+          if (isPlugin || channelID === '753291485100769411') {
+            const addonURL = args[0].message?.embeds[0]?.fields[0].rawValue;
+            const addonID = addonURL.split('/').pop().toLowerCase();
+
+            const pluginsOrThemes = isPlugin ? 'plugins' : 'themes';
 
-        return res;
-      }
-    );
+            const addonIsInstalled = vizality.manager[
+              pluginsOrThemes
+            ].keys.includes(addonID);
+
+            res.props.children.push(
+              <>
+                <ContextMenu.Separator />
+                <ContextMenu.Group>
+                  <ContextMenu.Item
+                    label={`${addonIsInstalled ? 'Uninstall' : 'Install'} ${isPlugin ? 'Plugin' : 'Theme'
+                    }`}
+                    id="addon-installer"
+                    action={async () => {
+                      if (addonIsInstalled) {
+                        await vizality.manager[pluginsOrThemes].uninstall(
+                          addonID
+                        );
+                      } else {
+                        await vizality.manager[pluginsOrThemes].install(addonURL);
+                      }
+                    }}
+                  />
+                </ContextMenu.Group>
+              </>
```

Comment:
This context menu group is *veeeery* similar to the one for Powercord. Would it be possible to merge them?

Reply:
Not really, we can only have powercord plugins support, but not themes support without the plugin somehow renaming `powercord_manifest` to `manifest`
But if you mean to make it outside of the if statement and return to it when it has everything, then I'll try that